final_project.py

```diff
@@ -0,0 +1,369 @@
+#   exist in network.
+# 
+# Arguments: 
+#   network: the gamer network data structure 
+#   user_A:  a string with the name of the user the connection is from
+#   user_B:  a string with the name of the user the connection is to
+#
+# Return: 
+#   The updated network with the new connection added.
+#   - If a connection already exists from user_A to user_B, return network unchanged.
+#   - If user_A or user_B is not in network, return False.
+def add_connection(network, user_A, user_B):
+    if user_A in network and user_B in network:
+        if user_B not in get_connections(network, user_A):
+            
+            network[user_A][0].append(user_B)
+            
+            return network
+        else:
+            pass       
+    else:
+        return False
+# ----------------------------------------------------------------------------- 
+# add_new_user(network, user, games): 
+#   Creates a new user profile and adds that user to the network, along with
+#   any game preferences specified in games. Assume that the user has no 
+#   connections to begin with.
+# 
+# Arguments:
+#   network: the gamer network data structure
+#   user:    a string containing the name of the user to be added to the network
+#   games:   a list of strings containing the user's favorite games, e.g.:
+#            ['Ninja Hamsters', 'Super Mushroom Man', 'Dinosaur Diner']
+#
+# Return: 
+#   The updated network with the new user and game preferences added. The new user 
+#   should have no connections.
+#   - If the user already exists in network, return network *UNCHANGED* (do not change
+#     the user's game preferences)
+def add_new_user(network, user, games):
+   
+    if user not in network:
+        connections =[] #new users don't have any connections
+        network[user] = [connections,games]
+        return network
+    else:
+        return network
+        
+# ----------------------------------------------------------------------------- 
+# get_secondary_connections(network, user): 
+#   Finds all the secondary connections (i.e. connections of connections) of a 
+#   given user.
+# 
+# Arguments: 
+#   network: the gamer network data structure
+#   user:    a string containing the name of the user
+#
+# Return: 
+#   A list containing the secondary connections (connections of connections).
+#   - If the user is not in the network, return None.
+#   - If a user has no primary connections to begin with, return an empty list.
+# 
+# NOTE: 
+#   It is OK if a user's list of secondary connections includes the user 
+#   himself/herself. It is also OK if the list contains a user's primary 
+#   connection that is a secondary connection as well.
+def get_secondary_connections(network, user):
+    if user in network:
+        secondary = []
+        
+        for name in get_connections(network, user):
+            secondary += get_connections(network, name) 
+                
+        return list(set(secondary)) #remove duplicates
```

Comment:
good! another way to do this would be to start off with `secondary` as a `set` directly, and keep adding elements to that. in that case you don't need to convert it later. but either ways are ok